runelite-api/src/main/java/net/runelite/api/CollisionDataFlag.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018, Woox <https://github.com/wooxsolo>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.api;
+
+public final class CollisionDataFlag
+{
+	public static final int BREAK_LINE_OF_SIGHT_FULL = 0x20000;
+	public static final int BREAK_LINE_OF_SIGHT_EAST = 0x1000;
+	public static final int BREAK_LINE_OF_SIGHT_WEST = 0x10000;
+	public static final int BREAK_LINE_OF_SIGHT_NORTH = 0x400;
+	public static final int BREAK_LINE_OF_SIGHT_SOUTH = 0x4000;
+
+	// 0x100 for objects, 0x200000 for unwalkable tiles such as water etc,
+	// 0x40000 is very rare but Barbarian Assault cannon has it.
+	public static final int BREAK_MOVEMENT_FULL = 0x100 | 0x200000 | 0x40000;
+
+	public static final int BREAK_MOVEMENT_EAST = 0x8;
+	public static final int BREAK_MOVEMENT_WEST = 0x80;
+	public static final int BREAK_MOVEMENT_NORTH = 0x2;
+	public static final int BREAK_MOVEMENT_SOUTH = 0x20;
```

Comment:
I have some existing unreleased work on these:
```
int BLOCK_NORTH_WEST = 0x1; 
int BLOCK_NORTH = 0x2; 
int BLOCK_NORTH_EAST = 0x4; 
int BLOCK_EAST = 0x8; 
int BLOCK_SOUTH_EAST = 0x10;
int BLOCK_SOUTH = 0x20;
int BLOCK_SOUTH_WEST = 0x40;
int BLOCK_WEST = 0x80;
int OBJ = 0x100; 
int FLOORDECO = 0x40000;
int FLOOR = 0x200000;
```
Additionally if you shift any of the BLOCK_ masks by << 9  you get the mask for whether it blocks projectiles.

Reply:
```
int BLOCK_NORTH_WEST = 0x1; 
int BLOCK_NORTH_EAST = 0x4; 
int BLOCK_SOUTH_EAST = 0x10;
int BLOCK_SOUTH_WEST = 0x40;
```
What are these used for? I got the line of sight algorithm from Henke, so he has probably tested it more than I have, but from what I've seen the algorithm works fine without those flags.

Reply:
They are used by the server sided pathing algorithm. In earlier versions of RS2 pathing was done in the client, that's where they come from.